host/sgx/sgxquote.c

```diff
@@ -288,12 +324,24 @@ static void _load_quote_ex_library_once(void)
             _quote_ex_library.sgx_get_supported_att_key_id_num(&att_key_id_num);
         if (status != SGX_SUCCESS || att_key_id_num == 0)
         {
-            OE_TRACE_ERROR(
-                "_load_quote_ex_library_once() "
-                "sgx_get_supported_att_key_id_num() status=%d num=%d\n",
-                status,
-                att_key_id_num);
-            OE_RAISE(OE_QUOTE_PROVIDER_CALL_ERROR);
+            if (defined_sgx_use_in_process_quoting)
+            {
+                // raise the error as the user intends to use quote-ex library,
+                // which turns out to be unavailable
+                OE_TRACE_ERROR(
+                    "_load_quote_ex_library_once() "
+                    "sgx_get_supported_att_key_id_num() status=%d num=%d\n",
+                    status,
+                    att_key_id_num);
+                OE_RAISE(OE_QUOTE_PROVIDER_CALL_ERROR);
+            }
+            else
+            {
+                // try using dcap instead
+                OE_TRACE_INFO("Try using in-process quoting.");
+                _quote_ex_library.use_dcap_library_instead = true;
```

Comment:
Instead of setting `use_dcap_library_instead` to true, I suggest calling `OE_RAISE`(error) which sets result to error and jumps to 'done' label which sets `_quote_ex_library.load_result` to error. `_quote_ex_library.load_result` also determines the result of `_use_quote_ex_library()`.

Reasons for the suggestion:
1. The if block at line 336 calls `OE_RAISE`(error). It's better to be consistent.
2. 'done:' label has function cleaning code. It's better to not skip that.
3. `_quote_ex_library.load_result` indicates the result of loading quote-ex and accompanying libraries. Setting it to error seems logical here.

Reply:
Sounds right.